packages/common/src/converter/data-converter.ts

```diff
@@ -66,6 +67,11 @@ export interface DataConverter {
   fromPayloadsSync<T>(index: number, content?: Payload[] | null): T;
 }
 
+export const isValidDataConverter = (dataConverter: unknown): boolean =>
+  typeof dataConverter === 'object' &&
+  dataConverter !== null &&
+  ['toPayload', 'toPayloads', 'fromPayload', 'fromPayloads'].every((method) => method in dataConverter);
+
```

Comment:
What's this for? Wouldn't any valid converter just implement the `DataConverter` interface?

Reply:
Interfaces are not JS things and are not present at runtime.

Reply:
Right, maybe I wasn't clear - I would expect this to narrow the type the way it's being used. As in https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
It's only used for validating `dataConverterPath`: https://github.com/temporalio/sdk-typescript/pull/430/files#diff-04516c8c3fdd8a63a8a2620f597ad8ff29fd98b3ec501026337b9e93af73668eR252

Perhaps it would be helpful to change the name to `isImportedDataConverterValid`? Or `doesImportedDataConverterLookRight` 😄